ctru-rs/examples/audio-filters.rs

```diff
@@ -64,10 +65,10 @@ fn main() {
 
     // We create a buffer on the LINEAR memory that will hold our audio data.
     // It's necessary for the buffer to live on the LINEAR memory sector since it needs to be accessed by the DSP processor.
-    let mut audio_data1 = Box::new_in([0u8; AUDIO_WAVE_LENGTH], LinearAllocator);
+    let mut audio_data1: Box<[_], _> = Box::new_in([0u8; AUDIO_WAVE_LENGTH], LinearAllocator);
```

Comment:
I haven’t tried running the example locally, but why is it needed to specify the type notation? It looks pretty clear from the call.

Reply:
It's to read it as a slice, gotcha.

Reply:
Yeah, and unfortunately because of the way default type parameters work we can't just write `Box<[_]>` here (since that turns into `Box<[_], GlobalAllocator>` which is the wrong type for the right-hand side). 

We _could_ write the API slightly more generically to allow for `Box<[u8; N]>`, maybe something like this? https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cf676ded2ec8a48a4ca7be66bec97c70

It's a bit messy though so I think I'd probably prefer to leave it, even if it means callers have to do this kind of coercion to use arrays (and also, `Vec` should always work without needing the extra type annotation).